crates/fluvio-version-manager/src/command/install.rs

```diff
@@ -0,0 +1,285 @@
+
+            if res.status() == StatusCode::Ok {
+                let out_path = tmp_dir.path().join(&artf.name);
+                let mut file = File::create(&out_path).unwrap();
+                let mut buf = Cursor::new(
+                    res.body_bytes()
+                        .await
+                        .map_err(|err| Error::msg(err.to_string()))?,
+                );
+
+                std::io::copy(&mut buf, &mut file)?;
+                tracing::debug!(
+                    "Artifact downloaded: {} at {:?}",
+                    artf.name,
+                    out_path.display()
+                );
+                continue;
+            }
+
+            tracing::warn!(
+                "Failed to download artifact {}@{} from {}",
+                artf.name,
+                artf.version,
+                artf.download_url
+            );
+            self.notify_warn(format!(
+                "Failed to download artifact {}@{}",
+                artf.name.bold(),
+                artf.version.to_string().bold()
+            ));
+        }
+
+        Ok(())
+    }
+
+    /// Verifies downloaded artifacts checksums against the upstream checksums
+    async fn checksum_artifacts(&self, tmp_dir: &TempDir, pkgset: &PackageSet) -> Result<()> {
+        for artf in pkgset.artifacts.iter() {
+            let bin_path = tmp_dir.path().join(&artf.name);
+
+            if bin_path.is_file() {
+                let mut binary = File::open(bin_path)?;
+                let shasum = Self::shasum256(&binary)?;
+                let upstream_shasum = surf::get(&artf.sha256_url)
+                    .await
+                    .map_err(|err| Error::msg(err.to_string()))?
+                    .body_string()
+                    .await
+                    .map_err(|err| Error::msg(err.to_string()))?;
+
+                if shasum == upstream_shasum {
+                    Self::set_executable_mode(&mut binary)?;
+                } else {
+                    return Err(Error::msg(format!(
+                        "Artifact {} didnt matched upstream shasum {}. Skipping installation for this artifact...",
+                        artf.name, shasum
+                    )));
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Allocates artifacts in the FVM `pkgset` directory for future use.
+    /// Returns the path to the allocated version directory.
+    async fn allocate_artifacts(&self, tmp_dir: &TempDir, pkgset: &PackageSet) -> Result<PathBuf> {
+        let version_path = fvm_versions_path()?.join(&self.version.to_string());
+
+        create_dir(&version_path)?;
+
+        for artif in pkgset.artifacts.iter() {
+            copy(
+                tmp_dir.path().join(&artif.name),
+                version_path.join(&artif.name),
+            )?;
+        }
+
+        Ok(version_path)
+    }
+
+    /// Writes out the manifest for the specified `PackageSet`
+    async fn write_out_manifest(
```

Comment:
Separate manifest logic from here. This should be moved to Channel obj so can consolidate it's behavior

Reply:
This is only used by FVM and is not part of the `Channel` logic. Manifest uses `Channel` as value but `Channel` does not need `Manifest`